caikit/core/model_management/model_saver.py

```diff
@@ -0,0 +1,117 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This file defines the abstraction for handling the output of training a model"""
+# Standard
+from typing import Optional
+import abc
+import os
+import typing
+
+# Local
+from ..modules import ModuleBase
+from caikit.interfaces.common.data_model.stream_sources import PathReference
+
+T = typing.TypeVar("T")
+
+
+class ModelSaver(typing.Generic[T]):
+    """Generic-typed, abstract base for model saver"""
+
+    @abc.abstractmethod
+    def save_model(
+        self, model: ModuleBase, model_name: str, training_id: Optional[str]
+    ) -> typing.Any:
+        """Save the loaded model, based on this target's configuration.
+
+        Args:
+            model (ModuleBase): a loaded model to be saved
+            model_name (str): The user-provided name of this model
+            training_id (str | None): The globally unique id tracking the training run that
+                created this model
+
+        Returns:
+            Any: Some representation of where the model was saved. This could be a path on disk.
+
+        Raises:
+            Any appropriate exception if saving the model fails
+        """
+
+    @abc.abstractmethod
+    def save_path(self, model_name: str, training_id: Optional[str]) -> typing.Any:
```

Comment:
I think all of this "look up the save path" was added thinking we _might_ need it, but in reality we don't actually use it. Since the typing semantics are gorpy anyway, let's remove `save_path` and the return value from `save_model`. We can add this back in if needed down the road.